src/main/java/seedu/address/storage/DataDirectoryArticleStorage.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.storage;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.logging.Logger;
+
+import org.apache.commons.codec.binary.Base32;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.util.FileUtil;
+
+/**
+ * A class to access UserPrefs stored in the hard disk as a json file
+ */
+public class DataDirectoryArticleStorage implements ArticleStorage {
+
+    private Path directoryPath;
+    private Logger logger = LogsCenter.getLogger(DataDirectoryArticleStorage.class);
+
+    public DataDirectoryArticleStorage(Path directoryPath) {
+        this.directoryPath = directoryPath;
+    }
+
+    @Override
+    public Path getArticleDataDirectoryPath() {
+        return directoryPath;
+    }
+
+    @Override
+    public void addArticle(String url, byte[] articleContent) throws IOException {
+        Path targetPath = getArticlePath(url);
+
+        // Ensure data directory exists
+        FileUtil.createDirectory(directoryPath);
+
+        FileUtil.writeToFile(targetPath, articleContent);
+    }
+
+    /**
+     * Converts the given url to a filename that will be used to write to.
+     */
+    private String urlToFilename(String url) throws NoSuchAlgorithmException {
+        String lowercaseUrl = url.toLowerCase();
+
+        try {
+            // We hash the URL with sha-256, truncate it to 128 bits so it's shorter, then encode it in base32
```

Comment:
this feels like overkill....

Reply:
We need some way of shortening the URL to reduce chances of hitting the 260 character path limit on Windows. Using truncated sha-256 is a pretty good and fast way of ensuring that. Finally we also need to encode in base32 in order to store the resultant hash in a way that doesn't conflict with the filesystem.